017-storage-buckets.md

```diff
@@ -0,0 +1,230 @@
+- **maximumFileSize = null** Maximum file size.
+- **allowedFileExtensions = null** Allowed File extensions.
+
+#### `DELETE: /v1/storage/bucket/{bucketId}`
+
+This endpoint deletes a Bucket by its unique ID. *Server*
+
+### Changes to existing Endpoints
+
+Since the first release of Buckets will not introduce a breaking change - the Buckets will be implement aside the current Storage service. Meaning the Buckets will be an additional feature to the existing API.
+
+#### `POST: /v1/storage/files`
+
+This endpoint creates a new file. This endpoint will have a 4th optional argument added called `bucket`.
+
+#### `GET: /v1/storage/files/{fileId}`
+#### `GET: /v1/storage/files/{fileId}/preview`
+#### `GET: /v1/storage/files/{fileId}/download`
+#### `GET: /v1/storage/files/{fileId}/view`
+
+These endpoints gets a file by its unique ID. These endpoints will a optional argument added called `bucket`.
+
+
+### Prior art
+
+[prior-art]: #prior-art
+
+<!--
+
+Discuss prior art, both the good and the bad, in relation to this proposal. A
+few examples of what this can include are:
+
+- Does this functionality exist in other software and what experience has their
+  community had?
+- For other teams: What lessons can we learn from what other communities have
+  done here?
+- Papers: Are there any published papers or great posts that discuss this? If
+  you have some relevant papers to refer to, this can serve as a more detailed
+  theoretical background.
+
+This section is intended to encourage you as an author to think about the
+lessons from other software, provide readers of your RFC with a fuller picture.
+If there is no prior art, that is fine - your ideas are interesting to us
+whether they are brand new or if it is an adaptation from other software.
+
+Write your answer below.
+-->
+- [About Storage Buckets](https://docs.uipath.com/orchestrator/docs/about-storage-buckets)
+- [Google Cloud Storage](https://cloud.google.com/storage/docs/json_api/v1/buckets)
+- [Google Cloud Storage - Key Terms](https://cloud.google.com/storage/docs/key-terms)
+
+### Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+<!-- What parts of the design do you expect to resolve through the RFC process before this gets merged? -->
+
+#### Prevent Breaking Changes
+
+To prevent a huge breaking change in the Storage Services API's - we are allowing users to use the current Storage implementation as well. Now there is the problem of offering a consistent API pattern across the services.
+
+Let's take a look at the `GET: /v1/storage/files/{fileId}` endpoint. This will result into retrieving a file from the current Storage service. Now if a file is inside a Bucket - the endpoint would have to look like `GET: /v1/storage/files/{bucketId}/{fileId}`. This becomes problematic, since right now it is not possible to allow optional Parameters in the Path with the same Controller/SDK Method and would require refactoring in all SDKs, Utopia and Appwrite.
```

Comment:
1. I like the endpoint `/v1/storage/files/{bucketId}/{fileId}
2. If no bucket is created may be appwrite can handle the bucket as `default` internally
3. Internally any path passed only as `/v1/storage/files/{fileId}` appwrite internally converts to `/v1/storage/files/default/{fileId}`